plugins/API/ProcessedReport.php

```diff
@@ -39,9 +40,8 @@ public function getMetadata($idSite, $apiModule, $apiAction, $apiParameters = ar
         $reportsMetadata = $this->getReportMetadata($idSite, $period, $date, $hideMetricsDoc, $showSubtableReports);
 
         foreach ($reportsMetadata as $report) {
-            // See ArchiveProcessor/Aggregator.php - unique visitors are not processed for period != day
-            // todo: should use SettingsPiwik::isUniqueVisitorsEnabled instead
-            if (($period && $period != 'day') && !($apiModule == 'VisitsSummary' && $apiAction == 'get')) {
+            // See ArchiveProcessor/Aggregator.php - unique visitors are not processed for all periods
+            if (($period && !SettingsPiwik::isUniqueVisitorsEnabled($period)) && !($apiModule == 'VisitsSummary' && $apiAction == 'get')) {
```

Comment:
In case you work on this PR again anyway it could be nice to make the code more readable and assign like meaningful variables. like `$isSummaryGetReport = $apiModule == 'VisitsSummary' && $apiAction == 'get';` and then in if `... !$isSummaryGetReport`